vault/audit.go

```diff
@@ -191,16 +195,19 @@ func (c *Core) disableAudit(ctx context.Context, path string, updateStorage bool
 	newTable := c.audit.shallowClone()
 	entry, err := newTable.remove(ctx, path)
 	if err != nil {
-		return false, err
+		return existed, err
 	}
 
 	// Ensure there was a match
 	if entry == nil {
-		return false, fmt.Errorf("no matching backend")
+		return existed, fmt.Errorf("no matching backend")
 	}
 
 	c.removeAuditReloadFunc(entry)
 
+	// We're satisfied that the entry exists now.
```

Comment:
👍  I would keep this comment though, because it deliberately shows the delineation between when `false` is returned as the existed return value and when `true` is returned.